new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleNginxValidation.java

```diff
@@ -628,4 +673,149 @@ private void checkServiceCreated(String serviceName) {
+
+        // check new managed server pods are ready
+        logger.info("Wait for the new managed server pod {0} to be ready in namespace {1}",
+            manageServerPodName, domainNamespace);
+        checkPodReady(manageServerPodName);
+
+        // check new managed server services are created
+        logger.info("Check for the new managed server service {0} is created in namespace {1}",
+            manageServerPodName, domainNamespace);
+        checkServiceCreated(manageServerPodName);
+
+        // add the new managed server to the list
+        expectedServerNamesInAppResponse.add(clusterName + "-managed-server" + i);
+
+      }
+
+      // check that NGINX can access the sample apps from new and original managed servers
+      logger.info("Check that NGINX can access the sample app from the new and original managed servers "
+          + "in the domain after the domain is scaled up.");
+      assertThat(callWebAppAndCheckForServerNameInResponse(curlCmd, expectedServerNamesInAppResponse, 50))
+          .as("Check NGINX can access the sample app from all managed servers in the domain")
+          .withFailMessage("NGINX can not access the sample app from one or more of the managed servers")
+          .isTrue();
+    } else {
+      // scale down
+      // wait and check the pods are removed
+      for (int i = replicasBeforeScale; i > numberOfServers; i--) {
+        logger.info("Check managed server pod {0} is removed in namespace {1}",
+            manageServerPodNamePrefix + i, domainNamespace);
+        checkPodDoesNotExist(manageServerPodNamePrefix + i);
+        expectedServerNamesInAppResponse.remove(clusterName + "-managed-server" + i);
+      }
+
+      // check that NGINX can access the remaining managed server in the domain
+      logger.info("Check that NGINX can access the sample app from the remaining managed servers in the domain "
+          + "after the domain is scaled down.");
+      assertThat(callWebAppAndCheckForServerNameInResponse(curlCmd, expectedServerNamesInAppResponse, 50))
+          .as("Check NGINX can access the sample app from the remaining managed server in the domain")
+          .withFailMessage("NGINX can not access the sample app from the remaining managed server")
+          .isTrue();
+    }
+
+    return numberOfServers;
+  }
+
+  /**
+   * Generate the curl command to ping the sample app from the ingress controller.
+   *
+   * @param clusterName WebLogic cluster name which is the backend of the ingress
+   * @return curl command string
+   */
+  private String generateCurlCmd(String clusterName) {
+    String curlCmd =
+        String.format("curl --silent --show-error --noproxy '*' -H 'host: %s' http://%s:%s/sample-war/index.jsp",
+        domainUid + "." + clusterName + ".test", K8S_NODEPORT_HOST, nodeportshttp);
+    return curlCmd;
+  }
+
+  /**
+   * Generate an expected server name list which should be returned in the sample app response.
```

Comment:
this implies the full list is in one response - in that what you meant? don't you mean the list that you expect to see after looking at a (large) number of responses

Reply:
fixed